client/lib/desktop-listeners/index.js

```diff
@@ -138,7 +142,7 @@ const DesktopListeners = {
 	onSignout: function () {
 		debug( 'Signout' );
 
-		userUtilities.logout();
+		redirectToLogout( getCurrentUser( this.store.getState() ) );
```

Comment:
I see some Redux patterns here:
- there could be a selector called `getLogoutUrl( state, redirect )` that reads from `state.currentUser.user` and constructs the logout URL.
- `redirectToLogout` could be an action thunk that is dispatched without any parameters. It reads all it needs from state internally. Is the Redux store accessible at all places where `redirectToLogout` is called?

Reply:
I was planning to go that way, but the instance in `wpcom-xhr-wrapper` was the one that doesn't have access to the store and we can't dispatch it there, not without using the redux bridge. What do you think?

Reply:
I think we can remove the `wpcom-xhr-wrapper` condition or replace it with a simple reload. Especially now when the Desktop app is just a wrapper and doesn't use OAuth. The condition was originally added in #612, solving a situation in Desktop app that doesn't have UI to reload the page.

The condition detects an expired OAuth token. By default, OAuth tokens expire in 14 days.

On the other hand, we don't have any check like this for cookies. The Calypso app will be getting failed REST responses until you reload the page. And the `wp_api` cookies in iframe proxy expire in only 2 days. The main `wordpress_sec` cookie lives practically forever, expiring in 3 years.